### ISSUE TITLE
Endless recursive requests to server when server response error

### ISSUE THREAD

**Issue by armstrongli** [gophercloud/gophercloud#232]
Here's the output from client when server is not stable. Then client sends more requests until connection reset.

```
...
ng request: Successfully re-authenticated, but got error ex
ecuting request: Successfully re-authenticated, but got error executing request: Successfully re-authenticated, but got error executi
ng request: Successfully re-authenticated, but got error executing request: Successfully re-authenticated, but got error executing re
quest: Successfully re-authenticated, but got error executing request: Successfully re-authenticated, but got error executing request
: Successfully re-authenticated, but got error executing request: Put https://os-r-object.vip.*.*.com/v1/KEY_3e4c92efc6444df19
f7df401a2de4a29/backup: read tcp 10.147.160.18:36116->10.14.41.20:443: read: connection reset by peer
```

**Comment by armstrongli** [comment]
Fixed in PR: https://github.com/gophercloud/gophercloud/pull/233

**Comment by jrperritt** [comment]
See https://github.com/gophercloud/gophercloud/pull/233#issuecomment-274197054

**Comment by armstrongli** [comment]
@jrperritt Please reopen  this issue as the root cause is found and fixed in this PR: https://github.com/gophercloud/gophercloud/pull/604

**Comment by jrperritt** [comment]
Looking now

**Comment by jrperritt** [comment]
closed by #604
